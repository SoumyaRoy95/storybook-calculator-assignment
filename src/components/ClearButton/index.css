.clearbutton {
  background-color: cornsilk;
  border: 1px solid brown;
  color: grey;
  font-size: 18px;
  padding: 10px;
  text-align: center;
  width: 105px;
}
