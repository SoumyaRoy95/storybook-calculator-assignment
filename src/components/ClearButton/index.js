import React from "react";
import "./index.css";

const ClearButton = (props) => {
  const { label, ...rest } = props;
  return (
    <button className="clearbutton" {...rest}>
      {label}
    </button>
  );
};

export default ClearButton;
