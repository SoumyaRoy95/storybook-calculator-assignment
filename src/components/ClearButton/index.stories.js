import React from "react";
import ClearButton from "./index";

export default {
  title: "ClearButton",
  component: ClearButton,
};

export const Clear = () => <ClearButton label="clear" />;
