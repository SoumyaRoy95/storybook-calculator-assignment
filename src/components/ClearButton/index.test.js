import ClearButton from "./index";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  render(<ClearButton label="clear" />);
});

it("renders button correctly", () => {
  render(<ClearButton label="clear" />);
  expect(screen.getByRole("button")).toHaveTextContent("clear");
});

it("matches screenshot", () => {
  const file = renderer.create(<ClearButton label="clear" />).toJSON();
  expect(file).toMatchSnapshot();
});
