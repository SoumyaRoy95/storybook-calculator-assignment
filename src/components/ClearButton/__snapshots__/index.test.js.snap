// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches screenshot 1`] = `
<button
  className="clearbutton"
>
  clear
</button>
`;
